blocks/minesweeper/src/app.ts

```diff
@@ -0,0 +1,65 @@
+import { BlockElementBase } from "@blockprotocol/graph/custom-element";
+import { getRoots } from "@blockprotocol/graph/stdlib";
+import { PropertyValues } from "lit";
+// eslint-disable-next-line import/extensions
+import { unsafeHTML } from "lit/directives/unsafe-html.js";
+import { mine_sweeper } from "mine-sweeper-tag";
+
+import { BlockEntity } from "./types/generated/block-entity";
+
+const colsKey: keyof BlockEntity["properties"] =
+  "https://blockprotocol-molpob88k.stage.hash.ai/@ciaranm/types/property-type/number-of-columns/";
+const minesKey: keyof BlockEntity["properties"] =
+  "https://blockprotocol-molpob88k.stage.hash.ai/@ciaranm/types/property-type/number-of-mines/";
+
+const takeNumberOrDefault = (value: unknown, defaultValue: number) => {
+  if (
+    typeof value === "number" ||
+    (typeof value === "string" && !Number.isNaN(parseInt(value, 10)))
+  ) {
+    return value;
+  }
+  return defaultValue;
+};
+
+export class MinesweeperBlock extends BlockElementBase<BlockEntity> {
```

Comment:
Interesting, why did you go with this approach?

Reply:
Part of the motivation for this block was to use the custom element template. It's also easier to render custom elements using this template than React – React requires a special wrapper for custom elements.